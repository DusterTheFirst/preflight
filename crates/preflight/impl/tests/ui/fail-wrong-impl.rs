use preflight_impl::{avionics_harness, Avionics, Control, Sensors};

#[derive(Default)]
struct Controller;

trait Cool {}

#[avionics_harness]
impl Cool for Controller {}

fn main() {}
