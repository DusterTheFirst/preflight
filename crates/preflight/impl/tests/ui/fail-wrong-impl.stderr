error: expected a trait implementation of `Avionics`
  --> $DIR/fail-wrong-impl.rs:14:6
   |
14 | impl Cool for Controller {}
   |      ^^^^

warning: unused imports: `Avionics`, `Control`, `Sensors`
 --> $DIR/fail-wrong-impl.rs:1:40
  |
1 | use preflight_impl::{avionics_harness, Avionics, Control, Sensors};
  |                                        ^^^^^^^^  ^^^^^^^  ^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
